modules/fidelityBidAdapter.js

```diff
@@ -1,102 +1,85 @@
+import * as utils from 'src/utils';
+import {registerBidder} from 'src/adapters/bidderFactory';
+
+const BIDDER_CODE = 'fidelity';
+const BIDDER_SERVER = 'x.fidelity-media.com';
+export const spec = {
+  code: BIDDER_CODE,
+  isBidRequestValid: function(bid) {
+    return !!(bid && bid.params && bid.params.zoneid);
+  },
+  buildRequests: function(validBidRequests, bidderRequest) {
+    return validBidRequests.map(bidRequest => {
+      var server = bidRequest.params.server || BIDDER_SERVER;
 
-var FidelityAdapter = function FidelityAdapter() {
-  var FIDELITY_BIDDER_NAME = 'fidelity';
-  var FIDELITY_SERVER_NAME = 'x.fidelity-media.com';
+      const payload = {
+        from: 'hb',
+        v: '1.0',
+        requestid: bidRequest.bidderRequestId,
+        impid: bidRequest.bidId,
+        zoneid: bidRequest.params.zoneid,
+        floor: parseFloat(bidRequest.params.floor) > 0 ? bidRequest.params.floor : 0,
+        charset: document.charSet || document.characterSet,
+        defloc: utils.getTopWindowUrl(),
+        altloc: window.location.href,
+        subid: 'hb',
+        flashver: getFlashVersion(),
+        tmax: bidderRequest.timeout,
+      };
+      if (document.referrer) {
+        payload.referrer = document.referrer;
+      }
 
-  function _callBids(params) {
-    var bids = params.bids || [];
-    bids.forEach(function (currentBid) {
-      var server = currentBid.params.server || FIDELITY_SERVER_NAME;
-      var m3_u = window.location.protocol + '//' + server + '/delivery/hb.php?';
-      m3_u += 'callback=window.$$PREBID_GLOBAL$$.fidelityResponse';
-      m3_u += '&requestid=' + utils.getUniqueIdentifierStr();
-      m3_u += '&impid=' + currentBid.bidId;
-      m3_u += '&zoneid=' + currentBid.params.zoneid;
-      m3_u += '&cb=' + Math.floor(Math.random() * 99999999999);
-      m3_u += document.charset ? '&charset=' + document.charset : (document.characterSet ? '&charset=' + document.characterSet : '');
-
-      var loc;
-      try {
-        loc = window.top !== window ? document.referrer : window.location.href;
-      } catch (e) {
-        loc = document.referrer;
-      }
-      loc = currentBid.params.loc || loc;
-      m3_u += '&loc=' + encodeURIComponent(loc);
-
-      var subid = currentBid.params.subid || 'hb';
-      m3_u += '&subid=' + subid;
-      if (document.referrer) m3_u += '&referer=' + encodeURIComponent(document.referrer);
-      if (currentBid.params.click) m3_u += '&ct0=' + encodeURIComponent(currentBid.params.click);
-      m3_u += '&flashver=' + encodeURIComponent(getFlashVersion());
-
-      adloader.loadScript(m3_u);
+      return {
+        method: 'GET',
+        url: '//' + server + '/delivery/hb.php',
+        data: payload
+      };
     });
-  }
+  },
+  interpretResponse: function(serverResponse) {
+    const bidResponses = [];
+    if (serverResponse && serverResponse.seatbid) {
+      serverResponse.seatbid.forEach(seatBid => seatBid.bid.forEach(bid => {
```

Comment:
Need to add required field `ttl` : Time-to-Live - how long (in seconds) Prebid can use this bid

Reply:
Ok, thanks for review. I'll add .md and update adapter by the end of today hopefully.

Reply:
@jaiminpanchal27 All done. Changes were made, checks passed.